src/api/Notes/notesApi.tsx

```diff
@@ -0,0 +1,74 @@
+import "@pnp/sp/webs";
+import "@pnp/sp/items";
+import "@pnp/sp/comments/item";
+import { subWebContext } from "src/api/SPWebContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { z } from "zod";
+
+const spNoteItem = z.array(
+  z.object({
+    Title: z.string().max(255, "Title must be 255 characters or less"),
+    Id: z.number(),
+  })
+);
+
+type spNoteItem = z.infer<typeof spNoteItem>;
+
+const useNoteItem = (program: string, pcolId: number) => {
+  return useQuery({
+    queryKey: ["noteItem", pcolId],
+    queryFn: () => getNoteItem(program, pcolId),
+    select: selectNoteId,
+    staleTime: Infinity, // Prevent refetch
+    gcTime: Infinity, // Prevent garbage collection
+  });
+};
+
+const getNoteItem = async (program: string, pcolId: number) => {
+  return subWebContext(program)
+    .web.lists.getByTitle("notes")
+    .items.filter(`Title eq ${pcolId}`)<spNoteItem>();
+};
+
+const selectNoteId = (noteItems: spNoteItem) => spNoteItem.parse(noteItems)[0];
+
+export const useNotes = (program: string, pcolId: number) => {
+  const noteItem = useNoteItem(program, pcolId);
+  const noteId = noteItem.data?.Id;
+
+  const hasNoteError = noteItem.error instanceof Error;
+  const returnFunction =
+    hasNoteError || !noteId
+      ? () => Promise.reject(noteItem.error) // If we erred trying to get noteItem, then return that error
```

Comment:
This doesn't quite work as expected. Since SharePoint isn't erroring if it can't find a record with that pcolID as the Title, it is just returing an empty array. Therefore, there is no noteItem.error object to pass to the reject, so the error is not thrown to the interface, and it makes the "Add Note" button available.